pkg/manifest/subscription.go

```diff
@@ -0,0 +1,72 @@
+package manifest
+
+import (
+	"github.com/osbuild/images/pkg/customizations/fsnode"
+	"github.com/osbuild/images/pkg/customizations/subscription"
+	"github.com/osbuild/images/pkg/osbuild"
+)
+
+type Subscription struct {
+	Base
+
+	Subscription *subscription.ImageOptions
+
+	// Custom directories and files to create in the pipeline
+	Directories []*fsnode.Directory
+	Files       []*fsnode.File
+}
+
+// NewSubscription creates a new subscription pipeline for creating files
+// required to register a system on first boot.
+// The pipeline is intended to be used to create the files necessary for
+// registering a system, but outside the OS tree, so they can be copied to
+// other locations in the tree after they're created (for example, to an ISO).
+func NewSubscription(buildPipeline Build, subOptions *subscription.ImageOptions) *Subscription {
+	name := "subscription"
+	p := &Subscription{
+		Base:         NewBase(name, buildPipeline),
+		Subscription: subOptions,
+	}
+	buildPipeline.addDependent(p)
+	return p
+}
+
+func (p *Subscription) serialize() osbuild.Pipeline {
+	pipeline := p.Base.serialize()
+	if p.Subscription != nil {
+		serviceDir, err := fsnode.NewDirectory("/etc/systemd/system", nil, nil, nil, true)
+		if err != nil {
+			panic(err)
+		}
+		p.Directories = append(p.Directories, serviceDir)
+
+		subStage, subDirs, subFiles, _, err := subscriptionService(*p.Subscription, &subscriptionServiceOptions{InsightsOnBoot: true, UnitPath: osbuild.EtcUnitPath})
+		if err != nil {
+			panic(err)
+		}
+		p.Directories = append(p.Directories, subDirs...)
+		p.Files = append(p.Files, subFiles...)
+
+		for _, stage := range append(osbuild.GenDirectoryNodesStages(p.Directories), osbuild.GenFileNodesStages(p.Files)...) {
+			if stage.Type == "org.osbuild.chown" {
+				// chown requires chrooting into the tree to run it, which
+				// wont work for this pipeline, so skip it
+				continue
+			}
```

Comment:
I have mixed feelings about this.

I have two suggestions:

1. Not set the `user` and `group` on files created by `subscriptionService()`, which will mean that there are no `chown` stages generated later. Keeping them `nil` will make them be owned by `root` implicitly AFAIK.
2. Extend the `Gen*NodesStages()` to optionally skip generation of `chown` stage.

I guess I could learn to live with the code as is in case you'd see some downside in both of my suggestions.

Reply:
Done.  Thanks!